From 0e28172db9e861006b729491264021048fd98ce2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:51 -0400
Subject: [usb] wusbcore: clean up the sg list that was created for out transfers

Message-id: <1410887463-20674-100-git-send-email-dzickus@redhat.com>
Patchwork-id: 95661
O-Subject: [RHEL7 PATCH 099/591] usb: wusbcore: clean up the sg list that was created for out transfers
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d5b5c9f228bf15d134b0ac8d493b119417f5f9e9
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Thu Sep 26 14:08:15 2013 -0500

    usb: wusbcore: clean up the sg list that was created for out transfers

    Clean up the SG list after transfer completetion for out transfers if one
    was created by the HWA.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-xfer.c b/drivers/usb/wusbcore/wa-xfer.c
index f614fb1..61b0597 100644
--- a/drivers/usb/wusbcore/wa-xfer.c
+++ b/drivers/usb/wusbcore/wa-xfer.c
@@ -520,6 +520,10 @@ static void wa_seg_dto_cb(struct urb *urb)
  unsigned rpipe_ready = 0;
  u8 done = 0;
 
+ /* free the sg if it was used. */
+ kfree(urb->sg);
+ urb->sg = NULL;
+
  switch (urb->status) {
  case 0:
   spin_lock_irqsave(&xfer->lock, flags);
-- 
1.7.1